### ISSUE TITLE
Used reward points are not displayed on order details pages or invoices (PDF)

### ISSUE THREAD

**Issue by RomanovM** [nopSolutions/nopCommerce#5115]
nopCommerce version : 4.3

Expected behavior : When Order is placed and reward point is used during purchase set ReedemedRewardPointEntryId on Order table.

Actual behavior: ReedemedRewardPointEntryId is Null on the Order table

Steps to reproduce the problem: 
- Place and order
- Use RewardPoint when placing an Order
- Reward points are not displayed on order details pages (admin area or public store). They are also not displayed in invoices (PDF)

Source: https://www.nopcommerce.com/en/boards/topic/88439/nopcommerce-43-bug-invoice-do-not-print-the-used-rewardpoints-on-the-invoice-both-email-and-pdf

**Comment by prasanthdevu** [comment]
Can we do any code changes to print this on invoice in version 4.3?

**Comment by Puzjak** [comment]
Hello. 
I also noticed this problem and found out that the issue is that NOP never saves Order.RedeemedRewardPointsEntryId in OrderProcessingService.SaveOrderDetails method and that property is used when preparing order details. I changed code from line 841 in OrderProcessingService.cs to look like this
`order.RedeemedRewardPointsEntryId = _rewardPointService.AddRewardPointsHistoryEntry(details.Customer, -details.RedeemedRewardPoints, order.StoreId, string.Format(_localizationService.GetResource("RewardPoints.Message.RedeemedForOrder", order.CustomerLanguageId), order.CustomOrderNumber), order, details.RedeemedRewardPointsAmount);
_customerService.UpdateCustomer(details.Customer);
_orderService.UpdateOrder(order);`
It now shows used reward points on order details, PDF, and mail.
